Source/Diagnostics/ReducedDiags/FieldProbe.cpp

```diff
@@ -0,0 +1,280 @@
+/* Copyright 2021 Lorenzo Giacomel
+ *
+ * This file is part of WarpX.
+ *
+ * License: BSD-3-Clause-LBNL
+ */
+
+#include "FieldProbe.H"
+#include "Particles/Gather/FieldGather.H"
+
+#include "Utils/CoarsenIO.H"
+#include "Utils/IntervalsParser.H"
+#include "WarpX.H"
+
+#include <AMReX_Array.H>
+#include <AMReX_Config.H>
+#include <AMReX_MFIter.H>
+#include <AMReX_MultiFab.H>
+#include <AMReX_ParallelDescriptor.H>
+#include <AMReX_ParmParse.H>
+#include <AMReX_REAL.H>
+#include <AMReX_Reduce.H>
+#include <AMReX_Geometry.H>
+
+#include <cmath>
+#include <ostream>
+
+using namespace amrex;
+
+// constructor
+FieldProbe::FieldProbe (std::string rd_name)
+: ReducedDiags{rd_name}
+{
+    // RZ coordinate is not working
+#if (defined WARPX_DIM_RZ)
+    AMREX_ALWAYS_ASSERT_WITH_MESSAGE(false,
+        "FieldProbe reduced diagnostics does not work for RZ coordinate.");
+#endif
+
+    // read number of levels
+    int nLevel = 0;
+    amrex::ParmParse pp_amr("amr");
+    pp_amr.query("max_level", nLevel);
+    nLevel += 1;
+
+    amrex::ParmParse pp_rd_name(rd_name);
+    pp_rd_name.query("x_probe", x_probe);
+    pp_rd_name.query("y_probe", y_probe);
+    pp_rd_name.query("z_probe", z_probe);
+
+    const bool raw_specified = pp_rd_name.query("raw_fields", raw_fields);
+    if(!raw_specified) raw_fields = false;
+
+    const bool interp_order_specified = pp_rd_name.query("raw_fields", interp_order);
```

Comment:
```suggestion
    const bool interp_order_specified = pp_rd_name.query("interp_order", interp_order);
```

Same comment as above also applies

Reply:
Also, to be extra safe, we could add a check that the selected interpolation order is not too large (otherwise segfaults or other weird things could happen). I think we could check that the allocated number of guard cells is at least equal to ceil(interp_order//2) (this should be verified) or more simply that interp_order <= WarpX::nox (more constraining).

Reply:
To keep it simple I'll go for the second one